Gobabygo/Gobabygo.ino

```diff
@@ -0,0 +1,74 @@
+const int joyStickXPin (0);
+// const int joyStickYPin (1);
+const int ledPin (6);
+// const int leftMotor (4);
+const float deadzone = 0.1;
+
+unsigned long currentTime = 0;
+unsigned long previousTime = 0;
+
+float joyXVal = 0;
+float scaledJoyXVal = 0.0;
+// int joyYVal = 0;
+// float scaledJoyYVal = 0.0;
+int ledVal = 0;
+float rightMotorVal = 0.0;
+float leftMotorVal = 0.0;
+
+
+void setup() {
+  // put your setup code here, to run once:
+  Serial.begin(9600);
+  pinMode(ledPin, OUTPUT);
+}
+
+void loop() {
+  // put your main code here, to run repeatedly:
+  joyXVal = analogRead(joyStickXPin);
+  scaledJoyXVal = joyXVal/1023*2 - 1;
+  if (abs(scaledJoyXVal) <= deadzone){
+    scaledJoyXVal = 0;
+  }
+  else {
+    if (scaledJoyXVal > deadzone) {
+      scaledJoyXVal = (scaledJoyXVal - 1)/(1 - deadzone) + 1; //modified point-slope form of a line
+    }
+    else {
+      scaledJoyXVal = (scaledJoyXVal + 1)/(1 - deadzone) - 1;
+    }
+  }
```

Comment:
It would be good to look into creating a method / function that encapsulated the logic for getScaledJoystickValue(int pin). This would be a method that reads and scales the joystick based on a given input pin. This would allow you to not have to duplicate logic for each joystick